extensions/jaeger/deployment/src/main/java/io/quarkus/jaeger/deployment/JaegerProcessor.java

```diff
@@ -22,13 +28,17 @@
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
     void setupTracer(JaegerDeploymentRecorder jdr, JaegerBuildTimeConfig buildTimeConfig, JaegerConfig jaeger,
-            ApplicationConfig appConfig, Capabilities capabilities) {
+            ApplicationConfig appConfig, Capabilities capabilities, BuildProducer<MetricBuildItem> metricProducer) {
 
         // Indicates that this extension would like the SSL support to be enabled
         extensionSslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(FeatureBuildItem.JAEGER));
 
         if (buildTimeConfig.enabled) {
-            boolean metricsEnabled = capabilities.isCapabilityPresent(Capabilities.METRICS);
+            boolean metricsEnabled = capabilities.isCapabilityPresent(Capabilities.METRICS)
+                    && buildTimeConfig.metricsEnabled;
+            if (metricsEnabled) {
```

Comment:
I'm pretty sure I've changed this code yesterday, you might want to rebase. (Wonder why GitHub doesn't complain...?)

Reply:
Git was able to handle the rebase itself, but the code after rebase was ugly (two `if` conditions after each other), so I fixed it up a bit

Reply:
You don't need this first `if` then, right? It seems duplicate.

Reply:
Hmm, yeah, seems I managed to mess it up. Fixed, thanks for noticing :D